internal/vault/client.go

```diff
@@ -429,9 +455,25 @@ func (c *defaultClient) startLifetimeWatcher(ctx context.Context) error {
 					logger.Error(err, "LifetimeWatcher completed with an error")
 					c.lastWatcherErr = err
 				}
+
+				c.watcher = nil
+				if c.watcherDoneCh != nil {
+					if !c.inClosing {
+						logger.V(consts.LogLevelTrace).Info("Writing to watcherDone channel")
+						c.watcherDoneCh <- c
+					} else {
+						logger.V(consts.LogLevelTrace).Info("In closing, not writing to watcherDone channel")
+					}
+				} else {
+					logger.V(consts.LogLevelTrace).Info("Skipping, watcherDone channel not set")
+				}
+
 				return
 			case renewal := <-watcher.RenewCh():
-				logger.Info("Successfully renewed the client")
+				logger.V(consts.LogLevelDebug).Info("Successfully renewed the client",
+					"cacheKey", cacheKey,
+					"accessor", renewal.Secret.Auth.Accessor)
```

Comment:
Is `renewal.Secret.Auth.Accessor` hashed at this point?

Reply:
Fixed in 1823da387328b5ecbbe6dee5a61ebcd0f1496834